helpers/getTestimonials.ts

```diff
@@ -1,40 +1,134 @@
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some((testimonialCategory) =>
+          categories.length > 0
+            ? categories.some(
+                (category) =>
+                  category === extractFileName(testimonialCategory.category)
+              )
+            : extractFileName(testimonialCategory.category) === "General"
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType)
+    ?.sort(() => 0.5 - Math.random())
+    ?.slice(0, 3);
+
+  return testimonialsResult;
+};
+
+export const getTestimonialsExcludingCategories = async (
+  categoryRoutePaths: {
+    categoryName: string;
+  }[]
+): Promise<TestimonialType[]> => {
+  const categoryListToExclude = categoryRoutePaths?.map(
+    (category) => extractFileName(category?.categoryName)?.toLowerCase()
+  );
+
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some(
+          (testimonialCategory) =>
+            categoryListToExclude.length > 0 &&
+            !categoryListToExclude.some(
+              (category) =>
+                category ===
+                extractFileName(testimonialCategory.category)?.toLowerCase()
+            )
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType);
+
+  return testimonialsResult;
+};
+export const getRandomClientTestimonial = () => {
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some(
+          (testimonialCategory) =>
+            SSWInternalTestimonialCategories.length > 0 &&
+            !SSWInternalTestimonialCategories.some(
+              (category) =>
+                category ===
+                extractFileName(testimonialCategory.category)?.toLowerCase()
+            )
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType);
+
+  const randomTestimonial =
+    testimonialsResult[Math.floor(Math.random() * testimonialsResult.length)];
+
+  return toTestimonialPanelObject(randomTestimonial);
+};
+
+export const testimonialToSelectOptions = () => {
+  const testimonialOptions = testimonialList.testimonials.map(
+    (testimonial) => ({
+      value: `${testimonial.name.toLocaleLowerCase()}`,
+      label: `${testimonial.name} - ${testimonial.company ?? ""}`,
+    })
+  );
+
+  return testimonialOptions;
+};
+
+export const getTestimonialByName = (name: string) => {
+  if (!name) return null;
+
+  const testimonialResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.name && testimonial.name.toLowerCase() === name
+    )
+    .map((testimonial) => testimonial as TestimonialType)[0];
+
+  return testimonialResult ? toTestimonialPanelObject(testimonialResult) : null;
```

Comment:
If there are no testimonials found - I'd expect the indexer to cause the code to error out.

Reply:
Done